cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/template/etl/common/ETLConfig.java

```diff
@@ -44,8 +45,8 @@ public ETLStage getSink() {
     return sink;
   }
 
-  public List<ETLStage> getTransforms() {
-    return transforms;
+  public Optional<List<ETLStage>> getTransforms() {
+    return Optional.fromNullable(transforms);
```

Comment:
Do we need to return an Optional? Can it be just `return transforms == null ? new ArrayList<>() : transforms;`?